### PR TITLE
Setup Codacy coverage push

### LINKED ISSUE #10: Improve code quality
Would you be interested in setting up Codacy to help improve the quality of this project? 

It can report on various quality, security, aspects whilst also feeding it coverage data from the tests that I have started in #8 

I can help set it up if you want.

### PR DESCRIPTION
**Pull request by inverse**
## Proposed change

Partially resolves #10 

@valayDave you'll still need to add the integration and setup the API key in the secrets called `CODACY_PROJECT_TOKEN`

## How to test the change

```
pytest --cov-report xml --cov=tell_me_your_secrets test/
```

Should generate a `coverage.xml` file

## Checklist

-   [ ] Tests have been added to verify that the new code works (if possible)
-   [ ] Documentation has been updated to reflect changes
-   [ ] Changelog has been updated to reflect changes

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,7 +26,13 @@ jobs:
         python -m pip install --upgrade pip
         pip install -r requirements-test.txt
     - name: Test with pytest
-      run: pytest
+      run: pytest --cov-report xml --cov=tell_me_your_secrets test/
+    - name: Upload coverage
+      if: ${{ matrix.python-version == 3.9 && env.IS_COVERAGE_ALLOWED == 'true' && env.IS_MASTER_BRANCH == 'true' }}
```
Comment (inverse) [inline .github/workflows/ci.yml:31]: Just so we don't run it on all versions and only when the secret + on master

**Comment by valayDave** [comment]
I was facing issues setting up the Token. Tried allocating the repo multiple times but it didn't work :( . Can you guide me docs.

**Comment by inverse** [comment]
Hi @valayDave what issues are you facing? 

From experience setting this up on my own repos I had to:

- login to app.codacy.com
- Choose my personal account
- Click "Add Repository"
- Enable the GitHub application
- Navigate back and add again

Once added and some initial checks are done you should be able to setup there should be a section on the bottom about code coverage which will take you to another page which should give you an API key which will need to be saved to the settings of this repo (like how pypi was setup)

**Comment by valayDave** [comment]
@inverse: Thanks for the steps. When I earlier did it the repository hadn't appeared. Now it did. Got the token and have set it up in the repository. Coverage is showing up in the CI.

**Comment by inverse** [comment]
@valayDave awesome! What's the next steps?

**Comment by valayDave** [comment]
Oh, even I don't know. I thought you just integrate this as normal and coverage reports will show up. No?

**Comment by inverse** [comment]
@valayDave enabling on the repo should provide some insights - do you have a URL to the page on codacy?

But this PR enables the coverage report to be pushed to their platform.

**Comment by valayDave** [comment]
@inverse: Apologies for the delay in response. 

This seems the Codacy URL: https://app.codacy.com/gh/valayDave/tell-me-your-secrets/dashboard

I think we can now integrate right ?

**Review by valayDave — APPROVED** [review]

**Comment by inverse** [comment]
image

Worked

**Pull request merged**